code/datums/quirks/negative.dm

```diff
@@ -982,3 +982,134 @@
+	lose_text = "<span class='notice'>You feel young again.</span>"
+	medical_record_text = "Patient suffers from a lack of mobility without a cane as support."
+	mail_goodies = list(/obj/item/cane)
+	var/datum/weakref/left_cane
+	var/datum/weakref/right_cane
+
+/datum/quirk/old_man_jenkins/add_unique()
+	var/mob/living/carbon/human/human_holder = quirk_holder
+
+	var/turf/holder_turf = get_turf(human_holder)
+	var/obj/item/cane/spawn_cane = new /obj/item/cane(holder_turf)
+	human_holder.put_in_hands(spawn_cane)
+
+	var/obj/item/left_hand_item = human_holder.held_items[1]
+	var/obj/item/right_hand_item = human_holder.held_items[2]
+	var/is_cane_left_hand = istype(left_hand_item) && HAS_TRAIT(left_hand_item, TRAIT_CANE_TOOL)
+	var/is_cane_right_hand = istype(right_hand_item) && HAS_TRAIT(right_hand_item, TRAIT_CANE_TOOL)
+	var/is_holding_cane = is_cane_left_hand || is_cane_right_hand
+
+	if(!is_holding_cane)
+		human_holder.add_movespeed_modifier(/datum/movespeed_modifier/bad_back_slowdown) // TODO make this it's own modifier at some point
+		quirk_holder.AddElement(/datum/element/waddling)
+	else if(is_holding_cane)
+		if(is_cane_left_hand)
+			left_cane = WEAKREF(left_hand_item)
+			RegisterSignal(left_hand_item, COMSIG_ITEM_DROPPED, .proc/on_unequipped_cane)
+			RegisterSignal(left_hand_item, COMSIG_PARENT_QDELETING, .proc/on_unequipped_cane)
+			RegisterSignal(left_hand_item, COMSIG_MOVABLE_MOVED, .proc/on_unequipped_cane)
+		if(is_cane_right_hand)
+			right_cane = WEAKREF(right_hand_item)
+			RegisterSignal(right_hand_item, COMSIG_ITEM_DROPPED, .proc/on_unequipped_cane)
+			RegisterSignal(right_hand_item, COMSIG_PARENT_QDELETING, .proc/on_unequipped_cane)
+			RegisterSignal(right_hand_item, COMSIG_MOVABLE_MOVED, .proc/on_unequipped_cane)
+
+	RegisterSignal(human_holder, COMSIG_MOB_EQUIPPED_ITEM, .proc/on_equipped_item)
+
+/datum/quirk/old_man_jenkins/remove()
+	UnregisterSignal(quirk_holder, list(COMSIG_MOB_EQUIPPED_ITEM, COMSIG_MOB_UNEQUIPPED_ITEM))
+	var/obj/item/primary_cane = left_cane?.resolve()
+	var/obj/item/secondary_cane = right_cane?.resolve()
+
+	var/obj/item/left_hand_item = quirk_holder.held_items[1]
+	var/obj/item/right_hand_item = quirk_holder.held_items[2]
+	var/is_cane_left_hand = istype(left_hand_item) && HAS_TRAIT(left_hand_item, TRAIT_CANE_TOOL)
+	var/is_cane_right_hand = istype(right_hand_item) && HAS_TRAIT(right_hand_item, TRAIT_CANE_TOOL)
+	var/is_holding_cane = is_cane_left_hand || is_cane_right_hand
+
+	if(!is_holding_cane)
+		quirk_holder.remove_movespeed_modifier(/datum/movespeed_modifier/bad_back_slowdown)
+		quirk_holder.RemoveElement(/datum/element/waddling)
+	else if(is_holding_cane)
+		if(primary_cane)
+			UnregisterSignal(primary_cane, list(COMSIG_ITEM_DROPPED, COMSIG_PARENT_QDELETING, COMSIG_MOVABLE_MOVED))
+		if(secondary_cane)
+			UnregisterSignal(secondary_cane, list(COMSIG_ITEM_DROPPED, COMSIG_PARENT_QDELETING, COMSIG_MOVABLE_MOVED))
+
+	left_cane = null
+	right_cane = null
```

Comment:
You don't need to do this btw, weakrefs are PODs and GC.

Reply:
What is PODs?  I know GC is garbage collection.

Reply:
No more need for weakrefs since I refactored it to use movement signal in https://github.com/tgstation/tgstation/pull/71470/commits/f3f3eadc10f674e64f41057130ed7eefa74ec715